lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3639,6 +3639,29 @@ namespace {
             return nullptr;
           }
         }
+
+        for (const auto method : methods) {
+          if (const auto clangDecl = method->getClangDecl()) {
+            const auto cxxMethod = dyn_cast<clang::CXXMethodDecl>(clangDecl);
+            if (cxxMethod &&
+                cxxMethod->isOverloadedOperator() &&
+                cxxMethod->getOverloadedOperator() == clang::OO_Subscript) {
+              assert(method->getParameters()->size() == 1);
+
+              const auto parameterType = method->getParameters()->get(0)->getType();
+              if (parameterType &&
+                  Impl.CXXSubscripts.find(parameterType) != Impl.CXXSubscripts.end()) {
+
+                auto cxxSubscripts = Impl.CXXSubscripts[parameterType];
+                auto subscript = makeSubscript(cxxSubscripts.first, cxxSubscripts.second);
+                result->addMember(subscript);
+
+                // Make sure we don't generate the same subscript twice.
+                Impl.CXXSubscripts.erase(parameterType);
```

Comment:
What is the case where this would happen?

Reply:
It would happen every time we read a struct with both const & non-const `operator[]`, since they are mapped into two different Swift methods. Added a comment to clarify that.